### PR TITLE
Add ms rest azure 0.11.1

### PR DESCRIPTION
**Pull request by apuntamb**
Supported Versions:

 * Nightly
 * 1.23
 * 1.22
 * 1.21
@ekohl Hi, here I'm creating separate PRs for set of dependencies required for the `foreman_azure_rm` plugin since we faced time out issues for the jobs in [#4032](https://github.com/theforeman/foreman-packaging/pull/4032). This is the first out of those to start with since `ms_rest_azure` has the largest number of dependencies.

### CODE REVIEW

**Comment by ekohl** [comment]
[test rpm]

**Comment by ekohl** [comment]
It's missing dependencies:
```
[2019-10-09T11:09:51.283Z]     package: tfm-rubygem-ms_rest-0.7.4-1.el7.noarch from downloaded_rpms
[2019-10-09T11:09:51.283Z]       unresolved deps:
[2019-10-09T11:09:51.283Z]          tfm-rubygem(timeliness) < 0:0.4
[2019-10-09T11:09:51.283Z]     package: tfm-rubygem-ms_rest_azure-0.11.1-1.el7.noarch from downloaded_rpms
[2019-10-09T11:09:51.283Z]       unresolved deps:
[2019-10-09T11:09:51.283Z]          tfm-rubygem(unf_ext) = 0:0.0.7.2
```

Looks like the timeliness version you're packaging is too new and `unf_ext` is missing from this PR.

**Comment by apuntamb** [comment]
@ekohl koji task fails for `unf_ext` now. Any suggestions?

File: packages/plugins/rubygem-unf_ext/rubygem-unf_ext.spec [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+mkdir -p %{buildroot}%{gem_extdir_mri}
+cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}/
+
+# Prevent dangling symlink in -debuginfo (rhbz#878863).
+rm -rf %{buildroot}%{gem_instdir}/ext/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/ext
```
Comment (kgaikwad) [inline packages/plugins/rubygem-unf_ext/rubygem-unf_ext.spec]: @apuntamb,
Koji build failed with below error related to `unf_ext` - 

```

Processing files: tfm-rubygem-unf_ext-0.0.7.2-1.el7.x86_64
BUILDSTDERR: warning: File not found: /builddir/build/BUILDROOT/tfm-rubygem-unf_ext-0.0.7.2-1.el7.x86_64/opt/theforeman/tfm/root/usr/share/gems/gems/unf_ext-0.0.7.2/ext
BUILDSTDERR: error: File not found: /builddir/build/BUILDROOT/tfm-rubygem-unf_ext-0.0.7.2-1.el7.x86_64/opt/theforeman/tfm/root/usr/share/gems/gems/unf_ext-0.0.7.2/ext
RPM build errors:
BUILDSTDERR:     File not found: /builddir/build/BUILDROOT/tfm-rubygem-unf_ext-0.0.7.2-1.el7.x86_64/opt/theforeman/tfm/root/usr/share/gems/gems/unf_ext-0.0.7.2/ext
BUILDSTDERR:     File not found: /builddir/build/BUILDROOT/tfm-rubygem-unf_ext-0.0.7.2-1.el7.x86_64/opt/theforeman/tfm/root/usr/share/gems/gems/unf_ext-0.0.7.2/ext
Child return code was: 1
EXCEPTION: [Error()]
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mockbuild/trace_decorator.py", line 96, in trace
    result = func(*args, **kw)
  File "/usr/lib/python2.7/site-packages/mockbuild/util.py", line 634, in do
    raise exception.Error("Command failed: \n # %s\n%s" % (command, output), child.returncode)
Error: Command failed: 
 # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/rubygem-unf_ext.spec
```
I think, you will need this `%{gem_instdir}/ext`.

Comment (kgaikwad) [reply packages/plugins/rubygem-unf_ext/rubygem-unf_ext.spec]: As per line 71 i.e `rm -rf %{buildroot}%{gem_instdir}/ext/`, you can actually delete these lines 75 & 80.

@ekohl, what do you think?

Comment (apuntamb) [reply packages/plugins/rubygem-unf_ext/rubygem-unf_ext.spec]: @kgaikwad Deleting lines- 75 and 80, works.
@ekohl Could you please take a look?

**Comment by apuntamb** [comment]
@ekohl Removing both lines [75](https://github.com/theforeman/foreman-packaging/commit/b752131ebf258c39401a63e40f002742bc5ce610#diff-40a692b316a476b191a15c775a5f0e5eR75) and [80](https://github.com/theforeman/foreman-packaging/commit/b752131ebf258c39401a63e40f002742bc5ce610#diff-40a692b316a476b191a15c775a5f0e5eR80) from the older commit was GREEN for us.
But, is failing when only line: 75 (as above) is removed.

**Comment by pcreech** [comment]
@apuntamb \o  looks like rubygem-unf_ext already exists.  Instead of adding a new one, mind bumping the existing one instead?

**Comment by apuntamb** [comment]
> @apuntamb \o looks like rubygem-unf_ext already exists. Instead of adding a new one, mind bumping the existing one instead?

@pcreech Ohkk! So, as per [ms_rest_azure](https://rubygems.org/gems/ms_rest_azure), it has a hard dependency on `unf_ext 0.0.7.2`. Should I bump up to the same or it might affect any other package?

**Comment by evgeni** [comment]
@apuntamb we don't require `unf_ext` directly, but only `unf`, and that has an open dependency on `unf_ext`, so it should be fine I think.
It strikes me odd tho, that the dependency in `ms_rest_azure` is `= 0.0.7.2`, that's oddly specific. Do you happen to know why this is so strict? Especially as there are newer versions of `unf_ext` available that should be compatible.

Looking at https://github.com/Azure/azure-sdk-for-ruby/pull/2457/files which points at https://github.com/meew0/discordrb/issues/604 this was due to missing ruby 2.6 builds for windows, which now seems to be fixed in 0.0.7.6?

**Comment by apuntamb** [comment]
> @apuntamb we don't require `unf_ext` directly, but only `unf`, and that has an open dependency on `unf_ext`, so it should be fine I think.
> It strikes me odd tho, that the dependency in `ms_rest_azure` is `= 0.0.7.2`, that's oddly specific. Do you happen to know why this is so strict? Especially as there are newer versions of `unf_ext` available that should be compatible.
**I remember earlier it had a dependency on `unf_ext 0.0.7.6` and when I'd created #4032 I didn't have to add `unf_ext` for the same reason that you mentioned above that we depend on `unf` and not directly on `unf_ext` which didn't give me any issue earlier based on this dependency.**
> 
> Looking at https://github.com/Azure/azure-sdk-for-ruby/pull/2457/files which points at [meew0/discordrb#604](https://github.com/meew0/discordrb/issues/604) this was due to missing ruby 2.6 builds for windows, which now seems to be fixed in 0.0.7.6?
> 
> Edit: I've just looked at our CI, and we run tests with unf_ext 0.0.7.6 anyways, so it should be fine.

**Comment by evgeni** [comment]
> Does this mean `unf_ext` is not required here at all (update/add)?
> 
> I remember earlier it had a dependency on `unf_ext 0.0.7.6` and when I'd created #4032 I didn't have to add `unf_ext` for the same reason that you mentioned above that we depend on `unf` and not directly on `unf_ext` which didn't give me any issue earlier based on this dependency.

By "should be fine" I meant that it should be fine to update `unf_ext` to 0.0.7.2 (or 0.0.7.6 if `ms_rest_azure` would allow that) as we run our tests against that version (indirectly, as `unf` pulls that in) anyways and they pass.

**Comment by apuntamb** [comment]
@evgeni @pcreech I tried to run `./bump_rpm.sh rubygem-unf_ext` but, it failed with this:
**rubygem-unf_ext is not a directory. It must be the full path**.

**Comment by pcreech** [comment]
@apuntamb I think you need to provide it a full relative path to the package.  Try running it with 

        ./bump_rpm.sh packages/foreman/rubygem-unf_ext/

File: packages/foreman/rubygem-unf_ext/rubygem-unf_ext.spec [hunk 2]
```diff
@@ -77,13 +77,17 @@ cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}
...
 %doc %{gem_instdir}/README.md
+%doc %{gem_instdir}/CHANGELOG.md
 %{gem_instdir}/%{gem_name}.gemspec
 %{gem_instdir}/test
 %{gem_instdir}/Gemfile
 %{gem_instdir}/Rakefile
 %exclude %{gem_instdir}/.*
 
 %changelog
+* Fri Oct 18 2019 vagrant <vagrant> 0.0.7.2-1
```
Comment (ekohl) [inline packages/foreman/rubygem-unf_ext/rubygem-unf_ext.spec]: Could you look at this?

Comment (apuntamb) [reply packages/foreman/rubygem-unf_ext/rubygem-unf_ext.spec]: done.

**Comment by apuntamb** [comment]
> Could you rebase on `rpm/develop` as well?

done.

**Comment by pcreech** [comment]
@apuntamb Feel free to open PR's against other versions as required

**Comment by evgeni** [comment]
@pcreech new packages require a `koji add-pkg` ;)

```
% koji add-pkg foreman-plugins-nightly-rhel7 tfm-rubygem-faraday-cookie_jar tfm-rubygem-ms_rest tfm-rubygem-ms_rest_azure tfm-rubygem-timeliness --owner=kojiadm             
Package tfm-rubygem-ms_rest already exists in tag foreman-plugins-nightly-rhel7
Package tfm-rubygem-ms_rest_azure already exists in tag foreman-plugins-nightly-rhel7
Adding 2 packages to tag foreman-plugins-nightly-rhel7
```

**Comment by evgeni** [comment]
It's now built in https://ci.theforeman.org/job/foreman-packaging-release/377, thanks @apuntamb!

**Pull request merged**